src/rcc.rs

```diff
@@ -456,27 +503,73 @@ impl CFGR {
             while rcc.csr.read().lsirdy().bit_is_clear() {}
         }
 
-        // Turn on HSI48 if required
-        if self.hsi48 {
+        if self.msi != MsiFreq::NOMSI {
+            unsafe { rcc.cr.modify(|_, w| w.msirange().bits(self.msi as u8).msirgsel().set_bit().msion().set_bit() )};
+            // Wait until MSI is running
+            while rcc.cr.read().msirdy().bit_is_clear() {}
+        }
+
+        (Hertz(hclk), Hertz(pclk1), Hertz(pclk2), ppre1, ppre2, Hertz(sysclk))
+    }
+
+
+    #[cfg(not(feature = "stm32l47x"))]
+    pub fn freeze(self, acr: &mut ACR) -> Clocks {
+
+        let (hclk, pclk1, pclk2, ppre1, ppre2, sysclk) = self.common_freeze(acr);
+        let mut usb_rng = false;
```

Comment:
Am I right in saying these extra freeze function just set `usb_rng`? If so what is useful about knowing usb_rng is enabled? I think most peripherals need to know what is providing the clock, so it can set its source appropriately.

Reply:
No the both freeze set the usb_rng boolean. The idea here is because of the RNG. The Rng need a clock source at 48Mhz (proven by stm to work correctly) (usb and rng clock share the same line IIUC).  But on some mcu you do have HSI48 that once enable will clock the rng source by default. On the other hand if you do not have HSI48 you can always back to the PLL48 clock source or the MSI possibly other clock source but that need to respect the 48Mhs constrain. So I can only enable the usb_rng with MSI (clk48sel register) if the MSI is 48Mhz. 

To conclude I need to feature gate this 2 branch in separate freeze as it have to take 2 different decision depending on the HSI48 being available or not.